scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -764,11 +764,46 @@
                                        {:job-uuid->reserved-host (apply dissoc job-uuid->reserved-host matched-job-uuids)
                                         :launched-job-uuids (into matched-job-uuids launched-job-uuids)})))
 
+(defn distribute-task-requests-to-compute-clusters
+  "Given a collection of un-matched task requests and a collection of
+  compute clusters, distributes the task requests amongst the compute
+  clusters, using a hash of the pending job's uuid."
+  [task-requests compute-clusters]
+  (group-by (fn [{:keys [job]}]
+              (nth compute-clusters
+                   (-> job :job/uuid hash (mod (count compute-clusters)))))
+            task-requests))
+
+(let [pool-name->last-autoscale-atom (atom {})]
+  (defn trigger-autoscaling!
```

Comment:
This function is low on logs, like when the iteration starts, when autocsaling is invoked on a cluster and when we are done with an iteration. We should also add metrics for these, e.g. time spent by each compute cluster in autoscaling. Number of jobs sent to each compute cluster, etc.

Reply:
Good call, will do

Reply:
Added logs and metrics